generator/src/main/java/com/scottlogic/deg/generator/restrictions/TextualRestrictions.java

```diff
@@ -0,0 +1,346 @@
+package com.scottlogic.deg.generator.restrictions;
+
+import com.scottlogic.deg.generator.generation.NoStringsStringGenerator;
+import com.scottlogic.deg.generator.generation.RegexStringGenerator;
+import com.scottlogic.deg.generator.generation.StringGenerator;
+import com.scottlogic.deg.generator.utils.SetUtils;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class TextualRestrictions implements StringRestrictions {
+    final Integer minLength;
+    final Integer maxLength;
+    final Set<Integer> excludedLengths;
+    final Set<Pattern> matchingRegex;
+    final Set<Pattern> containingRegex;
+    final Set<Pattern> notMatchingRegex;
+    final Set<Pattern> notContainingRegex;
+    private StringGenerator generator;
+
+    TextualRestrictions(
+        Integer minLength,
+        Integer maxLength,
+        Set<Pattern> matchingRegex,
+        Set<Pattern> containingRegex,
+        Set<Integer> excludedLengths,
+        Set<Pattern> notMatchingRegex,
+        Set<Pattern> notContainingRegex) {
+        this.minLength = minLength;
+        this.maxLength = maxLength;
+        this.matchingRegex = matchingRegex;
+        this.containingRegex = containingRegex;
+        this.excludedLengths = excludedLengths;
+        this.notMatchingRegex = notMatchingRegex;
+        this.notContainingRegex = notContainingRegex;
+    }
+
+    @Override
+    public boolean match(String x) {
+        return createGenerator().match(x);
+    }
+
+    /**
+     * Produce a new string restrictions instance that represents the intersection of this and the other given restrictions
+     * See MatchesStandardStringRestrictions.intersect() for more details on how the `aValid` operator can be merged
+     */
+    public StringRestrictions intersect(StringRestrictions other){
+        if (other instanceof NoStringsPossibleStringRestrictions){
+            return other;
+        }
+
+        if (other instanceof MatchesStandardStringRestrictions){
+            return other.intersect(this);
+        }
+
+        TextualRestrictions textualRestrictions = (TextualRestrictions) other;
```

Comment:
safe assumption to make?

Reply:
Currently this is the only remaining restriction, but could be guarded more explicitly

Reply:
Resolved